### DISCUSSION TITLE
[2주차] PUT과 PATCH의 차이점

### DISCUSSION THREAD

**Discussion by yunjjeongjo** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#33] [Interview Question]
PUT과 PATCH의 차이점에 대해서 설명해 주세요!

**Comment by sa02045** [comment]
- PUT 메서드는 요청으로 서버 자원의 `모든 필드`를 대체합니다. 전체 필드를 요청 body에 넣어야합니다. 그렇지 않고 누락된 필드가 있다면 서버는 이를 null로 처리해야합니다.

- PATCH 메서드는 요청으로 서버 자원의 `일부 필드`만 수정합니다. 일부 필드만을 요청 body에 넣어도 됩니다. 누락된 필드가 있어도 서버는 요청의 일부 필드로만 자원을 업데이트합니다.

PUT은 오로지 요청에 의해서만 전체 자원이 결정되기 때문에 `멱등성`을 가집니다. PATCH는 일부 필드만을 업데이트하기 때문에 멱등성을 가지지 않습니다.

**Reply by punchdrunkard** [reply]
멱등성 관점에서는 생각해본적이 없었는데 짚어주셔서 정말 감사합니다👍👍

**Reply by Hyevvy** [reply]
멱등성 부분 관련하여 벨로그 글 있어서 첨부합니다! 
좋은 답변 감사합니다!🙂🙂🙂

<img width="933" alt="스크린샷 2023-01-24 오후 8 42 13" src="https://user-images.githubusercontent.com/72402747/214282856-2db163cc-7e56-4dfe-b0c2-a3a2a52f42ed.png">

출처) [velog- 멱등성이 뭔가요](https://velog.io/@tosspayments/%EB%A9%B1%EB%93%B1%EC%84%B1%EC%9D%B4-%EB%AD%94%EA%B0%80%EC%9A%94)

**Reply by punchdrunkard** [reply]
**DB의 상태로 확인한다**
